src/components/organisms/ProfileModal/EditProfileLinkForm/EditProfileLinkForm.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useCallback, useMemo } from "react";
+import { useForm } from "react-hook-form";
+import * as Yup from "yup";
+
+import { updateProfileLinks } from "api/profile";
+
+import { ProfileLink, User } from "types/User";
+
+import { WithId } from "utils/id";
+import { isDefined } from "utils/types";
+
+import Button from "components/atoms/Button";
+import { InputField } from "components/atoms/InputField";
+
+import { UserProfileMode } from "../ProfilePopoverContent";
+
+import "./EditProfileLinkForm.scss";
+
+export interface EditProfileLinkFormProps {
+  edit?: boolean;
+  user?: WithId<User>;
+  profileLink?: ProfileLink;
+  setUserProfileMode: (mode: UserProfileMode) => void;
+}
+
+export const EditProfileLinkForm: React.FC<EditProfileLinkFormProps> = ({
+  edit = false,
+  user,
+  profileLink,
+  setUserProfileMode,
+}) => {
+  const title = edit ? "Edit your profile link" : "Add a profile link";
+  const saveButtonText = edit ? "Save changes" : "Add link";
+
+  const validateUniqueTitle = useCallback(
+    (title) => {
+      const allowedCountOfSameTitles = edit ? 1 : 0;
+      const sameTitleCount =
+        user?.profileLinks?.filter(
+          (userProfileLink) =>
+            userProfileLink.title === profileLink?.title ||
+            userProfileLink.title === title
+        ).length ?? 0;
+
+      return sameTitleCount === allowedCountOfSameTitles;
+    },
+    [profileLink, user?.profileLinks, edit]
+  );
+
+  const validationSchema = useMemo(
+    () =>
+      Yup.object().shape<ProfileLink>({
+        title: Yup.string()
+          .required("Title is required")
+          .test("unique title", "Title must be unique", validateUniqueTitle),
+        url: Yup.string()
+          .required("URL is required")
+          .matches(/^https?:\/\/\w+/, "URL must be valid"),
+      }),
+    [validateUniqueTitle]
+  );
+
+  const { register, handleSubmit, errors } = useForm<ProfileLink>({
+    validationSchema,
+    mode: "onBlur",
+    reValidateMode: "onChange",
+  });
+
+  const onSubmit = useCallback(
+    async (profileLinkFormData: ProfileLink) => {
+      if (!isDefined(user)) return;
+
+      const newProfileLinks = user.profileLinks ?? [];
+      const profileLinkToEdit = newProfileLinks.find(
+        (userProfileLink) => userProfileLink.title === profileLink?.title
+      );
+
+      if (profileLinkToEdit) {
+        Object.assign(profileLinkToEdit, profileLinkFormData);
+      } else {
+        newProfileLinks.push(profileLinkFormData);
+      }
```

Comment:
It looks too complicated. And I don't understand what is the intent here.

Reply:
Also, Object.assign must re rewritten not to mutate object, but to create a new one

Reply:
ES6+ object spread syntax, never `Object.assign`

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals

Reply:
no-mutation update is here: https://github.com/sparkletown/sparkle/pull/1499/commits/4b9c4fee3c3c8ac3d75c4a18814995447bfa29da#diff-7e60858c69455cc177571d9c3680b6df13c67916f9827da273b5e0bf46ee9ee7R70-R72

```ts
      const newProfileLinks = [...(user.profileLinks ?? [])];
      const profileLinkToEdit = newProfileLinks.findIndex(
        (link) => link.title === profileLink?.title
      );

      if (profileLinkToEdit > -1) {
        newProfileLinks[profileLinkToEdit] = profileLinkFormData;
      } else {
        newProfileLinks.push(profileLinkFormData);
      }
```

`profileLinkToEdit` if for the check if there is an profile link we need to replace / update (and if so, which exactly)